mayastor/src/bdev/nexus/nexus_channel.rs

```diff
@@ -125,12 +125,12 @@ impl NexusChannelInner {
             .iter_mut()
             .filter(|c| c.state() == ChildState::Open)
             .for_each(|c| {
-                self.writers.push(
-                    BdevHandle::try_from(c.get_descriptor().unwrap()).unwrap(),
-                );
-                self.readers.push(
-                    BdevHandle::try_from(c.get_descriptor().unwrap()).unwrap(),
-                );
+                if let Ok(hdl) = c.handle() {
+                    self.readers.push(hdl);
+                } else {
+                    c.set_state(ChildState::Faulted(Reason::CantOpen));
+                    error!("failed to create handle for {}", c);
```

Comment:
Should this message be "failed to get handle..." as we aren't actually creating a handle?